File: template.py
#!/usr/bin/env python3

from os.path import dirname, realpath, join
import time


def transform_input(input_):
    input_ = input_.strip()
    # custom transform for the day

    return input_


def read_input():
    dir_path = dirname(realpath(__file__))
    with open(join(dir_path, "input.txt"), "r") as f:
        for line in f:
            yield transform_input(line)


def solve_part1(input_):
    pass


def solve_part2(input_):
    pass


def main():
    start_time = time.time()
    solve_part1(read_input())
    elapsed_time = (time.time() - start_time) * 1e3
    print(f"Time elapsed for part 1: {elapsed_time} ms")

    start_time = time.time()
    solve_part2(read_input())
    elapsed_time = (time.time() - start_time) * 1e3
    print(f"Time elapsed for part 2: {elapsed_time} ms")


if __name__ == "__main__":
    main()

File: template.cpp
#include <chrono>
#include <fstream>
#include <iostream>
#include <vector>

using namespace std;
using chrono::duration_cast;
using chrono::microseconds;
using chrono::steady_clock;

vector<string> read_input() {
  ifstream f;
  f.open("input.txt");

  vector<string> input;
  string line;

  while (getline(f, line)) {
    input.push_back(line);
  }

  return input;
}

void solve_part1(const vector<string>& input) {}

void solve_part2(const vector<string>& input) {}

int main() {
  auto input = read_input();
  auto st1 = steady_clock::now();
  solve_part1(input);
  cout << "(time taken part1: "
       << duration_cast<microseconds>(steady_clock::now() - st1).count()
       << "[us])\n";
  auto st2 = steady_clock::now();
  solve_part2(input);
  cout << "(time taken part2: "
       << duration_cast<microseconds>(steady_clock::now() - st2).count()
       << "[us])\n";

  return 0;
}
